features/56-login-with-custom-token/api.md

```diff
@@ -0,0 +1,160 @@
+## Feature Overview
+
+Allow user to integrate their authentication server to skygear server by logging in with custom token
+
+## Guides
+
+### Sample Codes (Client SDK)
+
+**JS**
+
+```js
+skygear.auth.loginWithCustomToken("<custom_token>")
+.then(function(skygearUser) {
+
+}).catch(function(error) {
+
+});
+```
+
+**iOS (swift)**
+
+```swift
+SKYContainer.default().auth.login(withCustomToken: "<custom_token>") { (user, error) in
+    if error != nil {
+        print("error login with custom token: \(error)")
+        return
+    }
+    print("sign up successful")
+    print("user record: \(user)")
+    print("email: \(user!["email"])")
+}
+```
+
+**Android (java)**
+
+```java
+skygear.getAuth().loginWithCustomToken("<custom_token>", new AuthResponseHandler() {
+    @Override
+    public void onAuthSuccess(Record user) {
+        Log.i("Skygear loginWithCustomToken", "onAuthSuccess: Got token: " + user.getAccessToken());
+    }
+
+    @Override
+    public void onAuthFail(Error error) {
+        Log.e("Skygear loginWithCustomToken", "onAuthFail: Reason: " + error.getMessage());
+    }
+});
+```
+
+### Sample Codes (User's server)
+
+1. Get the token secret from portal, and setup environment variable in your server `SKYGEAR_CUSTOM_TOKEN_SECRET`
+
+2. Generate create token
+
+**Python (With SDK)**
+
+```python
+from skygear.utils.auth import create_custom_token
+
+custom_token = create_custom_token("your server user id", profile)
+
+```
+
+**Rudy (Without SDK)**
+
+- Install [ruby-jwt](https://github.com/jwt/ruby-jwt)
+
+```ruby
+custom_token_secret = "Custom token secret from portal"
+user_id = "unique identifier of user in your server, string between 1-36 characters long"
+now_seconds = Time.now.to_i
+
+payload = {
+    :sub => user_id,
+    :iat => now_seconds,
+    :exp => now_seconds+(60*60), # expire within 1 hour
+    :skygear_profile => {}
+}
+
+JWT.encode payload, custom_token_secret, "HS256"
+```
+
+### List of APIs (Client)
+
+**py-skygear**
+
+```
+skygear.utils.auth.create_custom_token(principle_id, profile={})
+```
+
+**JS**
+
+```js
+skygear.auth.loginWithCustomToken(customToken: String): Promise<Record>
```

Comment:
JS will have the `createCustomToken` for cloud function.

Reply:
For consistency, I think I will create example for how to create jwt token in js. I think we don't have much use case that user have to create custom token in cloud function. In case they need, they still can use the js example